src/qt/receivecoinsdialog.h

```diff
@@ -44,6 +45,8 @@ class ReceiveCoinsDialog : public QDialog
 
     void setModel(WalletModel *model);
 
+    RecentRequestsTableModel* recentRequestsTableModel() const { return m_recent_requests_table_model; }
```

Comment:
nit: seems like returning a `const` pointer would be more correct, given that this is a const method

Perhaps negligible as this is only used in the test, it has no other impact:

```diff
diff --git a/src/qt/receivecoinsdialog.h b/src/qt/receivecoinsdialog.h
index e400129f7..5dc340aed 100644
--- a/src/qt/receivecoinsdialog.h
+++ b/src/qt/receivecoinsdialog.h
@@ -45,7 +45,7 @@ public:
 
     void setModel(WalletModel *model);
 
-    RecentRequestsTableModel* recentRequestsTableModel() const { return m_recent_requests_table_model; }
+    const RecentRequestsTableModel* recentRequestsTableModel() const { return m_recent_requests_table_model; }
 
 public Q_SLOTS:
     void clear();
diff --git a/src/qt/test/wallettests.cpp b/src/qt/test/wallettests.cpp
index c15b1c0d6..276df75f4 100644
--- a/src/qt/test/wallettests.cpp
+++ b/src/qt/test/wallettests.cpp
@@ -212,7 +212,7 @@ void TestGUI(interfaces::Node& node)
     // Check Request Payment button
     ReceiveCoinsDialog receiveCoinsDialog(platformStyle.get());
     receiveCoinsDialog.setModel(&walletModel);
-    RecentRequestsTableModel* requestTableModel = receiveCoinsDialog.recentRequestsTableModel();
+    const RecentRequestsTableModel* requestTableModel = receiveCoinsDialog.recentRequestsTableModel();
     assert(requestTableModel);
 
     // Label input
```

Reply:
> nit: seems like returning a `const` pointer would be more correct, given that this is a const method

More correct why? Those are orthogonal.

Reply:
Well, not in every case, but in many cases it would be incongruent if an instance was const and its member was not - for example, if I have a `const` counter object, but I exposed the current count in an externally modifiable way, then that would be incongruent IMO.

In this case, it's not necessary to expose members in a non-const way from a const method, so I would bias toward the more restrictive presentation.

Reply:
An alternative is a method that returns a bool rather than the member itself, since that is all that is needed to satisfy the test.